akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerTestSetupBase.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.impl.engine.server
+
+import scala.concurrent.duration.FiniteDuration
+
+import akka.actor.ActorSystem
+import akka.event.NoLogging
+import akka.util.ByteString
+
+import akka.stream.FlowMaterializer
+import akka.stream.scaladsl.{ Sink, Source, FlowGraph }
+import akka.stream.testkit.{ TestPublisher, TestSubscriber }
+
+import akka.http.impl.util._
+
+import akka.http.ServerSettings
+import akka.http.scaladsl.model.headers.{ ProductVersion, Server }
+import akka.http.scaladsl.model.{ HttpResponse, HttpRequest }
+
+abstract class HttpServerTestSetupBase {
+  implicit def system: ActorSystem
+  implicit def materializer: FlowMaterializer
+
+  val requests = TestSubscriber.manualProbe[HttpRequest]
+  val responses = TestPublisher.manualProbe[HttpResponse]
+
+  def settings = ServerSettings(system).copy(serverHeader = Some(Server(List(ProductVersion("akka-http", "test")))))
+
+  val (netIn, netOut) = {
+    val netIn = TestPublisher.manualProbe[ByteString]
```

Comment:
you are recommended to use the normal (auto) probe instead of the manual, is usually powerful enough and simplifies life.